source/common/upstream/load_balancer_impl.cc

```diff
@@ -507,8 +515,52 @@ bool ZoneAwareLoadBalancerBase::earlyExitNonLocalityRouting() {
   return false;
 }
 
-HostConstSharedPtr LoadBalancerBase::chooseHost(LoadBalancerContext* context) {
+bool LoadBalancerContextBase::validateOverrideHostStatus(Host::Health health,
+                                                         OverrideHostStatus status) {
+  switch (health) {
+  case Host::Health::Unhealthy:
+    return status & UnhealthyStatus;
+  case Host::Health::Degraded:
+    return status & DegradedStatus;
+  case Host::Health::Healthy:
+    return status & HealthyStatus;
+  }
+  return false;
+}
+
+HostConstSharedPtr LoadBalancerContextBase::selectOverrideHost(const HostMap* host_map,
+                                                               LoadBalancerContext* context) {
+  if (context == nullptr) {
+    return nullptr;
+  }
+
+  auto override_host = context->overrideHostToSelect();
+  if (!override_host.has_value()) {
+    return nullptr;
+  }
+
+  if (host_map == nullptr) {
+    return nullptr;
+  }
+
+  auto host_iter = host_map->find(override_host.value().first);
+  HostConstSharedPtr host = host_iter != host_map->end() ? host_iter->second : nullptr;
```

Comment:
I think doing an early return here and breaking up L549 into two if checks would make it a bit easier to follow

Reply:
get it.